mula/scheduler/app.py

```diff
@@ -164,6 +165,24 @@ def collect_metrics(self) -> None:
                         count,
                     )
 
+            current_process = psutil.Process(os.getpid())
+            cpu = current_process.cpu_percent(interval=0.1)
+            mem = current_process.memory_info()
+
+            self.ctx.metrics_cpu_usage.labels(
+                type="cpu",
+                unit="percent",
+            ).set(
+                cpu,
+            )
+
+            self.ctx.metrics_memory_usage.labels(
+                type="memory",
+                unit="bytes",
+            ).set(
+                mem.rss,
+            )
```

Comment:
If you use kubernetes you already get these metrics per container via kubelet cadvisor built-in. If you use VMs, you can install process-exporter (https://github.com/ncabatoff/process-exporter) to get these metrics.

So I am not really sure whether we should duplicate those metrics inside every OpenKAT service. And if we add it I think some people might prefer to be able to disable it to prevent duplicate metrics in their prometheus database. But we can also ask what people who run and want to monitor OpenKAT think...

Reply:
Yeah, good point and I agree. Indeed it was mainly added for development purposes and I'll make it optional as such.